src/extra_files.c

```diff
@@ -107,16 +119,12 @@ static void handle(const char *filename, bool is_dir, bool fix)
 }
 
 /* expect the start to end in /usr and be the absolute path to the root */
-int walk_tree(struct manifest *manifest, const char *start, bool fix)
+int walk_tree(struct manifest *manifest, const char *start, bool fix, const regex_t *whitelist)
```

Comment:
So you are saying that "whitelist" should be a global variable?

I personally don't consider that good style and tried to avoid it, hence the additional parameter to walk_tree(). Then for the callback there's no alternative, but at least it is static to extra_files.c and not global to all of swupd-client.

But I can change it.

Reply:
Thinking about this some more I still didn't like adding even more global variables to the entire program. This gets executes so infrequently (once per program run!) that the overhead is completely negligible.

Let me know whether this really needs to be changed.

Reply:
No, it doesn't need to be changed.